lib/katello/tasks/clean_backend_objects.rake

```diff
@@ -72,6 +72,15 @@ namespace :katello do
       end
     end
 
+    def host_unregister_options(host)
+      if host.managed? || host.compute_resource
+        print "Leaving #{host.name} record in place, it is either managed or assigned to a compute resource."
```

Comment:
I would say "Foreman record" just so people know the Pulp + Katello + Candlepin records are expected to disappear